docs/topics/multiplatform/multiplatform-hierarchy.md

```diff
@@ -1,111 +1,340 @@
-iOS targets. The new structure helps you share more native code without being limited by platform-specific dependencies.
+Starting with Kotlin 1.8.20, you can set up source set hierarchy in your multiplatform projects with default target hierarchy.
+It's a [template](#see-the-full-hierarchy-template) for all possible targets and their shared source sets hardcoded in the Kotlin Gradle plugin.
 
-By using the hierarchical structure along with platform-dependent libraries in shared source sets, you can eliminate the
-need to use workarounds to get IDE support for sharing source sets among several native targets, for
-example `iosArm64` and `iosX64`:
+### Set up your project
+
+To set up a hierarchy, call `targetHierarchy.default()` in the `kotlin` block of your `build.gradle(.kts)` and list all the targets you need.
+For example:
 
 ```kotlin
-// workaround 1: select iOS target platform depending on the Xcode environment variables
+@OptIn(ExperimentalKotlinGradlePluginApi::class)
 kotlin {
-    val iOSTarget: (String, KotlinNativeTarget.() -> Unit) -> KotlinNativeTarget =
-        if (System.getenv("SDK_NAME")?.startsWith("iphoneos") == true)
-            ::iosArm64
-        else
-            ::iosX64
+    // Enable the default target hierarchy:
+    targetHierarchy.default()
 
-    iOSTarget("ios")
+    android()
+    iosArm64()
+    iosSimulatorArm64()
 }
 ```
-{initial-collapse-state="collapsed"}
 
-```bash
-# workaround 2: make symbolic links to use one source set for two targets
-ln -s iosMain iosArm64Main && ln -s iosMain iosX64Main
+When you declare final targets `android`, `iosArm64`, and `iosSimulatorArm64` in your code, the Kotlin Gradle plugin finds
+suitable shared source sets from the template and creates them for you. The resulting hierarchy looks like this:
+
+![An example of using the default target hierarchy](default-hierarchy-example.svg){thumbnail="true" width="350" thumbnail-same-file="true"}
+
+Green source sets are actually created and present in the project, while gray ones from the default template are
+ignored. As you can see, The Kotlin Gradle plugin hasn't created the `watchos` source set, for example, because there
+are no watchOS targets in the project.
+
+If you add a watchOS target, like `watchosArm64`, the `watchos` source set is created, and the code
+from `apple`, `native`, and `common` source sets is compiled to `watchosArm64` as well.
+
+> In this example, `apple` and `native` source sets compile only to `iosArm64` and `iosSimulatorArm64` targets.
+> Therefore, despite their names, they have access to the full iOS API.
+> This might be counter-intuitive for source sets like `native`, as you may expect that only APIs available on all
+> native targets are accessible in this source set. This behavior may change in the future.
+>
+{type="note"}
+
+### Adjust resulting hierarchy
+
+If necessary, you can further configure the resulting hierarchy manually [using the `dependsOn` relation](#manual-configuration).
```

Comment:
It's not clear what actually "If necessary" means and how user can understand that it is necessary for their project.
Could we omit this?